src/tools/pub/glossary.md

```diff
@@ -145,34 +144,12 @@ A [dependency](#dependency) that your package directly uses itself. The
 dependencies you list in your pubspec are your package's immediate dependencies.
 All other dependencies are [transitive dependencies](#transitive-dependency).
 
-## Library package
+## Library
 
-A package that other packages can depend on. Library packages can have
-[dependencies](#dependency) on other packages *and* can be dependencies
-themselves. They can also include scripts to be run directly. The
-opposite of a library package is an [application package][].
+A library is a single compilation unit, made up of a single primary file and any
+optional number of [parts][]. Libraries have their own private scope.
 
-[application package]: #application-package
-[lockfile]: #lockfile
-[version constraints]: #version-constraint
-[immediate dependencies]: #immediate-dependency
-
-Don't check the [lockfile][] of a library package into source
-control, since libraries should support a range of dependency versions. The
-[version constraints][] of a library package's
-[immediate dependencies][] should be as wide as possible while still
-ensuring that the dependencies will be compatible with the versions that were
-tested against.
-
-Since [semantic versioning](https://semver.org/spec/v2.0.0-rc.1.html) requires
-that libraries increment their major version numbers for any backwards
-incompatible changes, library packages will usually require their dependencies'
-versions to be greater than or equal to the versions that were tested and less
-than the next major version. So if your library depended on the (fictional)
-`transmogrify` package and you tested it at version 1.2.1, your version
-constraint would be [`^1.2.1`][].
-
-[`^1.2.1`]: /tools/pub/dependencies#caret-syntax
+[parts]: /language/libraries#importing-only-part-of-a-library
```

Comment:
This doc is not describing part files, so I don't think its a good resource for this link. Do we have a better one?

Reply:
Thanks for catching that, changed it to https://dart.dev/effective-dart/usage#do-use-strings-in-part-of-directives. Another option is https://dart.dev/tools/diagnostic-messages#part-file but that's kind of sparse. Probably some consolidating that needs to happen there but I'll save that for another PR

Reply:
Those are both better - probably the latter one is best if it was filled out a bit.